src/lib/libast/include/ast_intercept.h

```diff
@@ -1,247 +0,0 @@
-*          Copyright (c) 1985-2013 AT&T Intellectual Property          *
-*                      and is licensed under the                       *
-*                 Eclipse Public License, Version 1.0                  *
-*                    by AT&T Intellectual Property                     *
-*                                                                      *
-*                A copy of the License is available at                 *
-*          http://www.eclipse.org/org/documents/epl-v10.html           *
-*         (with md5 checksum b35adb5213ca9657e911e9befb180842)         *
-*                                                                      *
-*              Information and Software Systems Research               *
-*                            AT&T Research                             *
-*                           Florham Park NJ                            *
-*                                                                      *
-*               Glenn Fowler <glenn.s.fowler@gmail.com>                *
-*                    David Korn <dgkorn@gmail.com>                     *
-*                     Phong Vo <phongvo@gmail.com>                     *
-*                                                                      *
-***********************************************************************/
-#pragma prototyped
-/*
- * ast function intercepts
- */
-
-#if !defined(_AST_INTERCEPT) && !_BLD_ast && _API_ast && _API_ast < 20130625 /* <ast_api.h> not in scope yet */
-#define _AST_INTERCEPT			0
-#endif
-
-#ifndef _AST_INTERCEPT
-#define _AST_INTERCEPT			1
-
-#define AST_SERIAL_ENVIRON		1
-#define AST_SERIAL_LOCALE		2
-#define AST_SERIAL_RESTART		3
-
-#define AST_SERIAL_get			0
-#define AST_SERIAL_inc			1
-#define AST_SERIAL_always		0xffffffff
-#define AST_SERIAL_except		(AST_SERIAL_always-1)
-#define AST_SERIAL_max			(AST_SERIAL_except-1)
-
-#ifndef _AST_INTERCEPT_IMPLEMENT
-#define _AST_INTERCEPT_IMPLEMENT	1
-
-/* at*() intercepts */
-
-#undef	faccessat
-#define faccessat	ast_faccessat
-
-#undef	fchmodat
-#define	fchmodat	ast_fchmodat
-
-#undef	fchownat
-#define fchownat	ast_fchownat
-
-#undef	fstatat
-#define fstatat		ast_fstatat
-
-#undef	linkat
-#define linkat 		ast_linkat
-
-#undef	mkdirat
-#define mkdirat		ast_mkdirat
-
-#undef	openat
-#define openat		ast_openat
-
-#undef	readlinkat
-#define readlinkat	ast_readlinkat
-
-#undef	renameat
-#define renameat 	ast_renameat
-
-#undef	symlinkat
-#define symlinkat	ast_symlinkat
-
-#undef	unlinkat
-#define unlinkat	ast_unlinkat
-
-/* restart intercepts */
-
-#undef	access
-#define access		ast_access
-
-#undef	chdir
-#define chdir		ast_chdir
-
-#undef	chmod
-#define chmod		ast_chmod
-
-#undef	chown
-#define chown		ast_chown
-
-#undef	close
-#define close		ast_close
-
-#undef	creat
-#define creat		ast_creat
-
-#undef	dup
-#define dup		ast_dup
-
-#undef	dup2
-#define dup2		ast_dup2
-
-#undef	eaccess
-#define eaccess		ast_eaccess
```

Comment:
Annoyingly macOS doesn't provide `eaccess()`. So while I agree that this intercept mechanism should be removed we do need a Meson feature test for `eaccess()` and a fallback implementation if it isn't available. Even better would be to eliminate the use of this function since it can result in security bugs. It's always better to just open the file and deal with that returning EACCESS.

Reply:
@krader1961 There are still some tests failing on FreeBSD. Mostly because of how ```/dev/fd``` behaves differently on it. I tried mounting ```fdescfs```, but still tests fail. I believe ce69c59 should not be required. Can you take a look at it ?

Reply:
> Can you take a look at it ?

Yes but it might take a couple of days since a) I'm switching to a new phone and b) the keyboard on my primary workstation is broken.